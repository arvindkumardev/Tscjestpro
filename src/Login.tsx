import React,{useState, useEffect} from "react";
import {View, Text, TextInput, TouchableOpacity, SafeAreaView, FlatList, StyleSheet} from "react-native";
import {useNavigation} from '@react-navigation/native';
// import Auth from "./Auth";
import { Auth } from '../src';
import constants from './locales/constants';


type InputTextProps = {
    setText?: any;
    placeholderText?: string;
    secureText?: boolean;
    keyboardEntry?: boolean;
  };
  

const Login = () => {
    const [email, setEmail] = useState<any>("");
    const [password, setPassword] = useState<any>("");
    const [list, setList] = useState<any>();
    const navigation = useNavigation<any>();

    useEffect(() => {
        fetch('https://jsonplaceholder.typicode.com/todos/1')
        .then(response => response.json())
        .then((data) => { setList(data)})
    }, [])

    return(
        <SafeAreaView style={style.mainView}>
        <View style={style.center}>
            <Text style={style.loginTopText}>{constants.Login}</Text>
            <TextInput 
            style={style.inputText}
            placeholder="Email"
            value={email}
            onChangeText={e => setEmail(e)}
            />
            <TextInput 
            style={style.passwordInput}
            placeholder="Password"
            secureTextEntry = {true}
            value={password}
            onChangeText={(password) => setPassword(password)}
            />
            <View style={style.row}>
            <Text style={style.donthaveText}>{constants.DontHaveAccount} </Text>
            <TouchableOpacity onPress={()=> navigation.navigate("Register")}>
            <Text style={style.registerHere}>{constants.RegisterHere}</Text>
            </TouchableOpacity>
            {/* <FlatList 
                 data={list}
                 keyExtractor={(listItem: listItemType) => {return listItem.id.toString();  }}
                 renderItem={({item})=> { return <Text style={{color: "green"}}>{item.title}</Text> } }
                /> */}
            </View>
           
            <TouchableOpacity style={style.loginView}
                onPress={()=> Auth.SignIn(email, password)}>
                <Text style={style.loginText}>{constants.Login}</Text>
                
                </TouchableOpacity>
                {/* <TouchableOpacity onPress={()=> navigation.navigate("ForgetPassword")}>
            <Text style={style.registerHere}>{constants.ResetPassword}</Text>
            </TouchableOpacity> */}
        </View>
        </SafeAreaView>
    );
}

const style = StyleSheet.create({
    mainView: {
        flex: 1, 
        justifyContent: 'center',
    },
    center: {
        alignItems: 'center',
    },
    loginTopText: {
        fontSize: 40, 
        fontWeight: "bold", 
        textAlign: 'center', 
        color: 'green',
    },
    inputText: {
        borderRadius: 5,
                backgroundColor: "white",
                width: "90%",
                height: 70,
                paddingLeft: 10,
                borderColor: "#000",
                borderWidth: 3,
                marginTop: 50,
    },
    passwordInput: {
        borderRadius: 5,
                backgroundColor: "white",
                width: "90%",
                height: 70,
                paddingLeft: 10,
                borderColor: "#000",
                borderWidth: 3,
                marginTop: 20,
    },
    row: {
        flexDirection: 'row'
    },
    donthaveText: {
        fontSize: 14, 
        marginTop: 10,
    },
    registerHere: {
        fontSize: 14, 
        marginTop: 10, 
        color: 'green',
    },
    loginView: {
        backgroundColor: "green",
                borderRadius: 5,
                width: "90%",
                height: 70,
                alignItems: "center",
                justifyContent: "center",
                marginTop: 80,
    },
    loginText: {
        color: '#fff',
    fontWeight: "bold",
    fontSize: 22,
    },
});
export default Login;