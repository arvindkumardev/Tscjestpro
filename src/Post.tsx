import React,{ useEffect } from "react";
import {View, Text, InteractionManager, StyleSheet} from 'react-native';
import { useRoute } from '@react-navigation/native';
import constants from "./locales/constants";

const Post = () => {
    const route = useRoute<any>();

    useEffect(()=> {
       const {Item} = route.params;
    }, [])
    return(
       <View style={{marginHorizontal: 10, marginTop: 10}}>
           <Text style={style.text}>{constants.Id} {route.params.Item.id}</Text>
           <Text style={style.text}>{constants.Title} {route.params.Item.title}</Text>
           <Text style={style.text}>{constants.Description} {route.params.Item.body}</Text>
       </View>
    );
}

const style = StyleSheet.create({
     text: {
        fontSize: 16, 
        color: "#000", 
        marginTop: 5,
     }
});
export default Post;