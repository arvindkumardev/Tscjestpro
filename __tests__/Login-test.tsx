/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import App from '../App';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 import Login from '../src/Login';
 import Post from '../src/Post';
 
 // it('renders correctly', () => {
 //   renderer.create(<App />);
 // });
 
 test('test render snapshot', () => {
   const snapshot = renderer.create(<Login />).toJSON();
   console.log("testing",snapshot);
   expect(snapshot).toMatchSnapshot();
 })
 