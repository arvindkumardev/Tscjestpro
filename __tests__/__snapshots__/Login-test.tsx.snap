// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "green",
          "fontSize": 40,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Login
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 5,
          "borderWidth": 3,
          "height": 70,
          "marginTop": 50,
          "paddingLeft": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#000",
          "borderRadius": 5,
          "borderWidth": 3,
          "height": 70,
          "marginTop": 20,
          "paddingLeft": 10,
          "width": "90%",
        }
      }
      value=""
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "marginTop": 10,
          }
        }
      >
        Don't have an account ?
         
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "green",
              "fontSize": 14,
              "marginTop": 10,
            }
          }
        >
          Register Here
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "green",
          "borderRadius": 5,
          "height": 70,
          "justifyContent": "center",
          "marginTop": 80,
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
